src/repositories/concrete/levelBlockchainRepository.ts

```diff
@@ -18,23 +18,27 @@ export class LevelBlockchainRepository implements IBlockchainRepository {
         throw new Error("Method not implemented.");
     }
 
-    public async getBlockAsync(blockNumber: number): Promise<Block> {
+    public async getBlocksByDatePeriodAsync(beginDate: Date, endDate: Date): Promise<Block[]> {
         throw new Error("Method not implemented.");
     }
 
-    public async setBlocksAsync(blocks: Block[]): Promise<void> {
+    public async getBlocksByHashAsync(blockHash: string): Promise<Block[]> {
+        throw new Error("Method not implemented.");
+    }
+
+    public async getBlockAsync(blockNumber: number): Promise<Block> {
         throw new Error("Method not implemented.");
     }
 
-    public async setBlockAsync(block: Block): Promise<void> {
+    public async getPreviousBlockAsync(parentHash: string): Promise<Block> {
         throw new Error("Method not implemented.");
     }
 
-    public async deleteBlocksAsync(blockNumbers: number[]): Promise<void> {
+    public async createBlocksAsync(blocks: Block[]): Promise<void> {
```

Comment:
Change `createBlocksAsync` to `addBlocksAsync` because it only adds the block to the data store instead of also creating a new block.